File: 30-07-2025_Longest_Subarray_With_Maximum_Bitwise_AND.py
class Solution:
    def longestSubarray(self, nums):
        maxi = max(nums)
        maxlen = 0
        len_count = 0
        
        for num in nums:
            if num == maxi:
                len_count += 1
                maxlen = max(maxlen, len_count)
            else:
                len_count = 0
        
        return maxlen

File: 30-07-2025_Longest_Subarray_With_Maximum_Bitwise_AND.java
class Solution {
    public int longestSubarray(int[] nums) {
        int maxi = 0;
        for (int num : nums) {
            if (num > maxi) {
                maxi = num;
            }
        }
        
        int maxlen = 0, len = 0;
        for (int num : nums) {
            if (num == maxi) {
                len++;
                maxlen = Math.max(maxlen, len);
            } else {
                len = 0;
            }
        }
        return maxlen;
    }
}
